blog/29-12-22-crowdfunding-using-hardhat/Create a Crowdfunding Smart Contract on Celo using Hardhat.md

```diff
@@ -0,0 +1,990 @@
+      await testERC.connect(user2).approve(raiseMoney.address, getFiveHundred)
+
+      // to give
+      await raiseMoney.connect(user1).give(1, getFifty);
+      await raiseMoney.connect(user2).give(1, getTwoHundred);
+
+      const ONE_MONTH_IN_SECS = 31 * 24 * 60 * 60;
+      const unlockTime = (await time.latest()) + ONE_MONTH_IN_SECS;
+
+      // Transactions are sent using the first signer by default
+      await time.increaseTo(unlockTime);
+
+      await expect(raiseMoney.connect(user1).refund(1)).not.to.be.reverted;
+    });
+
+  });
+});
+```
+
+Once you have compiled it, this should show on your terminal:
+![compile and test](images/test%20celo.jpeg)
+
+Now, we can move to the next phase, which is deployment to the Celo Alfajores Testnet.
+
+# Deployment of the Crowdfunding Contract to Celo
+
+Before you plan to deploy, ensure you delete the configuration file–which is the `Hardhat.config.js` in your Hardhat and replace it with this:
+
+```js
+require("@nomiclabs/hardhat-waffle");
+require("dotenv").config({ path: ".env" });
+require("hardhat-deploy");
+
+// Prints the Celo accounts associated with the mnemonic in .env
+task("accounts", "Prints the list of accounts", async (taskArgs, hre) => {
+  const accounts = await hre.ethers.getSigners();
+
+  for (const account of accounts) {
+    console.log(account.address);
+  }
+});
+
+/**
+ * @type import('hardhat/config').HardhatUserConfig
+ */
+module.exports = {
+  defaultNetwork: "alfajores",
+  networks: {
+    localhost: {
+      url: "http://127.0.0.1:7545",
+    },
+    alfajores: {
+      gasPrice: 1500000000,
+      gas: 4100000,
+      url: "https://alfajores-forno.celo-testnet.org",
+      accounts: {
+        mnemonic: process.env.MNEMONIC,
+        path: "m/44'/52752'/0'/0",
+      },
+      //chainId: 44787
+    },
+    celo: {
+      url: "https://forno.celo.org",
+      accounts: {
+        mnemonic: process.env.MNEMONIC,
+        path: "m/44'/52752'/0'/0",
+      },
+      chainId: 42220,
+    },
+  },
+  solidity: "0.8.0",
+};
+```
+
+Then deploy the contract with `hardhat run scripts/deploy.js --network alfajores`. This was the contract address of our contract: 0xEF87eDE6F4713De16d83Ca08cf0f8A6a263Db60A.
+
+![deploy](images/celo%20result.jpeg)
+
+You can always verify your contracts at [Celoscan](https://alfajores.celoscan.io/)
+
+# Conclusion
+
+Congratulations, you just learned how to build and deploy a crowdfunding contract on the Celo blockchain. This project should have enhanced your contract of not only smart contract development with Solidity but also the intricacies of successfully building on Celo.
+
+Ensure you take your time to code along in this tutorial, and push your code to GitHub when you are done.
+
+# Next Steps
+
+For more real-time visual testing, I recommend you test this contract on Remix, using up to 3 addresses to contribute to the crowdfunding contract.
+
+# About the Author
+
+John Fawole is a blockchain technical writer and Solidity dev.
```

Comment:
We ask authors to only share Github or Linkedin url. Feel free to add if you want.